packages/vm/src/middleware/deterministic.rs

```diff
@@ -138,14 +138,386 @@ impl FunctionMiddleware for FunctionDeterministic {
             | Operator::I64Extend8S
             | Operator::I64Extend16S
             | Operator::I64ExtendI32S
-            | Operator::I64ExtendI32U => {
+            | Operator::I64Extend32S
+            | Operator::I64ExtendI32U
+            | Operator::RefNull { .. }
+            | Operator::RefIsNull
+            | Operator::RefFunc { .. }
+            | Operator::ReturnCall { .. }
+            | Operator::ReturnCallIndirect { .. }
+            | Operator::TypedSelect { .. }
```

Comment:
I guess this referes to

> In future versions of WebAssembly, the type annotation on 𝗌𝖾𝗅𝖾𝖼𝗍 may allow for more than a single value being selected at the same time.

from https://webassembly.github.io/spec/core/syntax/instructions.html#syntax-instr-parametric

which is mentioned in https://github.com/WebAssembly/reference-types/blob/master/proposals/reference-types/Overview.md as well:

> The select instruction now optionally takes a value type immediate. Only annotated select can be used with reference types.